src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp

```diff
@@ -513,14 +511,15 @@ Array<Array<Access>> Generator::computeAccessesForFixedTargets(
     };
 
     const auto computeElevations =
-        [&zenithRotations, &fromPositionCoordinates_ITRF](const Vector3d& aToPositionCoordinates_ITRF) -> VectorXd
+        [&fromPositionCoordinates_ITRF](const Vector3d& aToPositionCoordinates_ITRF) -> VectorXd
     {
         const MatrixXd dx = (-fromPositionCoordinates_ITRF).colwise() + aToPositionCoordinates_ITRF;
-        const VectorXd dx_Z = zenithRotations.col(0).array() * dx.row(0).transpose().array() +
-                              zenithRotations.col(1).array() * dx.row(1).transpose().array() +
-                              zenithRotations.col(2).array() * dx.row(2).transpose().array();
+        const MatrixXd fromPositionDirection_ITRF = fromPositionCoordinates_ITRF.colwise().normalized();
 
-        return (dx_Z.transpose().array() / dx.colwise().norm().array()).asin();
+        const VectorXd dx_Z =
+            (dx.cwiseProduct(fromPositionDirection_ITRF)).colwise().sum().array() / dx.colwise().norm().array();
+
+        return dx_Z.array().asin();
```

Comment:
I imagine this works out to the same thing as the dot product between the two vectors in order to get the z axis projection, so you could add a comment saying so? Haven't seen the `cwiseProduct` method before.

Reply:
Yeah it's basically a dot product column wise for a matrix. Will add some notes. cwiseProduct is well explain on the eigen documentation.